src/header/header.scss

```diff
@@ -330,7 +330,7 @@
 
   > *:first-child {
     .iui-button {
-      padding-left: $iui-sm;
+      padding-left: $iui-s;
```

Comment:
idk.. for me 12 px is random number and should be used on edge cases. Maybe this is the one, cant say.

Reply:
> because the caret icon occupies invisible space.

He is correct, the icon takes up 16x16 but is visibly smaller than that.  In this screenshot I've added four 12px red blocks & one 16px green block to help demonstrate.
![Screen Shot 2022-04-12 at 4 34 57 PM](https://user-images.githubusercontent.com/849817/163049119-8dcc9074-f9a9-421b-897f-7b9239cc087f.png)

Reply:
These changes do feel a little cramped but not so much that I'd dismiss the suggestion. If the reason for the change is so we can show a 3rd header button within a target device screen width, then I'd say that change makes sense.  But if this is just for more variable consistency, I'm more hesitant.

I think we could solve [the original issue](https://github.com/iTwin/iTwinUI/issues/574#issuecomment-1090477211) a different way.

Reply:
Suggestions?

Reply:
> I think normal padding should be 12px on both sides, and only split-button should have less on the right

^My suggestion

Reply:
So when you say split-button, you also include dropdown button? Cause visually they look alike.

I still dont get the 12px part, though.
If I check our simple button, it has 16px padding, for example (on the left) and 8px on right.
![image](https://user-images.githubusercontent.com/81580355/163186577-c0077858-3d48-4611-9c29-d2425e7b625a.png)
Then if I check split button, left padding stays 16px, and 2nd button with arrow has only 4px right padding
![image](https://user-images.githubusercontent.com/81580355/163186785-18331427-4bc9-42e2-af30-73ef16487896.png)

Reply:
Oh, should have added this button instead
![image](https://user-images.githubusercontent.com/81580355/163187556-74abcfa5-d3fa-4875-aa76-f62412a9640d.png)

With icon default one has 16px left padding.
Small one has 8px.
So again, 12px is some custom stuff, which I might just not understand the reasoning for :/

Reply:
I think the reasoning for 12px vs 16px was just visual preference.  I tend to lean towards Mayank's suggestion as well:  leave the padding at 12px unless a dropdown is present (split or not).

Reply:
> So when you say split-button, you also include dropdown button?

Yeah I meant dropdown.

> If I check our simple button, it has 16px padding ...

> So again, 12px is some custom stuff, which I might just not understand the reasoning for

I agree 12px vs 16px is inconsistent. And it doesn't make sense to look at menu items because they are not aligned with our dropdown buttons. Maybe we need to find a common value that works for both default button and header buttons?
![image](https://user-images.githubusercontent.com/9084735/163233580-598f44bd-51cd-498b-9252-ac6b02bbcbab.png)

Reply:
Ok, I have created a new class for dropdown button and applied the needed padding.. Specificty needed to be higher to override header button, but.. Hopefully refactor will improve things there.
Now visuals hsould be as you wish, even though I am not a fan of those 🗡️ 

Also hpefully fixed menu alignment issue.

Reply:
> I have created a new class for dropdown button

Shouldn't this also apply to non-split dropdown buttons? So most of the images shouldn't change.

Reply:
not sure what you have meant with your comment, but I did forget adding new class in `menu`, `skip-to-content` html files.
I did that and few images got rid of.
Now all images are changed either because of additional new button I added in demo, or menu position.

Reply:
Yeah that makes sense. Ignore my comment 😅